flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/JobReconciler.java

```diff
@@ -109,6 +110,14 @@ public JobReconciler(
             }
         }
 
+        if (SavepointUtils.checkSavepointAnnotation(flinkApp)) {
+            triggerSavePoint(flinkApp, effectiveConfig);
```

Comment:
I think we need to an asynchronous way to do the savepoint since it will take little long time and block the operator thread.

Reply:
I think we can do the Asynch way pretty nicely, as the rest api allows async trigger/polling for the savepoint status. So the flow could be:

 1. If savepoint annotation is set to `trigger`
    - Send savepoint trigger to flink rest api
    - Set savepoint annotation to `in-progress/pending`
    - Store pending savepoint information (trigger id) to job status
 2. If savepoint annotation is set to `in-progress/pending` Observer should poll for savepoint trigger id
   - If finished, remove annotation, update last savepoint
   - If still pending, reschedule

Reply:
This is really a great idea. Big +1 for this.

One more thing, if the savepoint is in `in-progress/pending`, we should not schedule a new one. This could be done via `validator`.

Reply:
Ok, I'll work in these ideas today